### PR TITLE
feat: Allow CAIP-10 addresses in `Address`

### PR DESCRIPTION
**Pull request by FrederikBolding**
Allow CAIP-10 addresses in the `Address` component.

This will need a follow-up PR in the extension to support as well.

### CODE REVIEW

**Comment by FrederikBolding** [comment]
@metamaskbot update-pr

File: packages/snaps-sdk/src/jsx/components/Address.ts [hunk 1]
```diff
@@ -7,22 +9,26 @@ import { createSnapComponent } from '../component';
  * valid Ethereum address, starting with `0x`.
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/components/Address.ts:9]: This needs to be updated.

File: packages/snaps-sdk/src/jsx/components/Address.ts [hunk 2]
```diff
@@ -7,22 +9,26 @@ import { createSnapComponent } from '../component';
...
- * An address component, which is used to display an Ethereum address.
+ * An address component, which is used to display a CAIP-10 address or a Ethereum address.
  *
  * This component does not accept any children.
  *
  * @param props - The props of the component.
- * @param props.address - The (Ethereum) address to display. This should be a
- * valid Ethereum address, starting with `0x`.
+ * @param props.address - The address to display. This should be a
+ * valid Ethereum address, starting with `0x` or a valid CAIP-10 address.
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/components/Address.ts:24]:
```suggestion
 * valid Ethereum address, starting with `0x`, or a valid CAIP-10 address.
```

File: packages/snaps-sdk/src/jsx/components/Address.ts (hunk 1 above)
Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/components/Address.ts]: Fixed!

File: packages/snaps-sdk/src/jsx/components/Address.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/components/Address.ts]: Fixed!

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**